openwisp_monitoring/monitoring/base/models.py

```diff
@@ -362,9 +342,9 @@ def top_fields(self):
 
     @property
     def _default_query(self):
-        q = "SELECT {field_name} FROM {key} WHERE time >= '{time}'"
+        q = default_chart_query[0]
         if self.metric.object_id:
-            q += " AND content_type = '{content_type}' AND object_id = '{object_id}'"
+            q += default_chart_query[1]
```

Comment:
This doesn't look like enough abstraction and it's very tied to the influxdb query yet

Reply:
Any suggestions :smiley:

Reply:
looks better now